apps/client/src/AppRouter.tsx

```diff
@@ -42,9 +42,27 @@ export default function AppRouter() {
     if (!data) return;
 
     for (const d of data) {
-      if (`/${d.alias}` === location.pathname && d.enabled) {
-        navigate(`/${d.pathAndParams}`);
-        break;
+      if (location.search.indexOf('alias=') > -1) {
+        // if the alias fits the alias on this page, if the URL is diferent, we redirect user to the new URL
+        let aliasOnPage = location.search.substring(location.search.indexOf('alias=') + 6);
+        if (aliasOnPage.indexOf('&') > -1) {
+          // take care of scenario where there is an & after the alias
+          aliasOnPage = aliasOnPage.substring(0, aliasOnPage.indexOf('&'));
+        }
+        // if we have the same alias and its enabled and its not empty
+        if (d.alias !== '' && d.enabled && d.alias === aliasOnPage) {
+          let fullLocationPath = location.pathname + location.search;
+          // we need to remove the &alias=ALIAS_NAME or ?alias=ALIAS_NAME at the end of the URL
+          fullLocationPath = fullLocationPath.replaceAll(`alias=${d.alias}`, '');
+          fullLocationPath = fullLocationPath.substring(0, fullLocationPath.length - 1);
+          // if the location path is different
+          if (d.pathAndParams !== fullLocationPath) {
```

Comment:
I am unsure if we can guarantee that the alias is at the end of the URL. 

How about extracting the params using the react router hook (useSearchParams) and comparing the object to one made from the urlAndParams? We would need to build an object from the urlAndParams string and deeply compare both objects. We have a library for the comparison in the project